src/main/java/com/google/api/codegen/metacode/InitCodeNode.java

```diff
@@ -198,7 +198,23 @@ public static InitCodeNode createTree(InitCodeContext context) {
     }
 
     root.resolveNamesAndTypes(context, context.initObjectType(), context.suggestedName(), null);
-    root.resolveSampleParamConfigs(context, "");
+
+    for (Map.Entry<String, SampleParameterConfig> entry :
+        context.sampleParamConfigMap().entrySet()) {
+      Scanner scanner = new Scanner(entry.getKey());
+      InitCodeNode parent = FieldStructureParser.parsePath(root, scanner);
+      int token = scanner.lastToken();
+      if (token == Scanner.EOF) {
+        parent.resolveSampleParamConfig(context, entry.getValue());
+      } else if (token == '%') {
```

Comment:
This is a cool idea, but I feel it's a bit unnecessary for a simple one-layer `if-else if` statement.

Reply:
It's a style thing, and I'm certainly not going to block on this.

But I will say that I like that style, because you know that once you reach the end of that code block, you're done processing. You don't have to jump to the ending braces to see whether there's more processing about to happen for that case.

I agree that it's less pressing for smaller branches, although my own personal threshold is shrinking, I find: a case with more than ~2 lines, I like to move into a guard-clause style.